core/domain/app_feedback_report_services.py

```diff
@@ -0,0 +1,86 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services to operate on app feedback report models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+
+(app_feedback_report_models,) = models.Registry.import_models(
+    [models.NAMES.app_feedback_report])
+transaction_services = models.Registry.import_transaction_services()
+
+
+def scrub_report(report_id, scrubbed_by):
+    """Scrubs the instance of AppFeedbackReportModel with given ID, removing
+    any user-entered input in the entity.
+
+    Args:
+        report_id: str. The id of the model entity to scrub.
+        scrubbed_by: str. The id of the user or cron job that is intiating
+            scrubbing this report.
+    """
+    _scrub_report_in_transaction(report_id, scrubbed_by)
+
+
+@transaction_services.run_in_transaction_wrapper
+def _scrub_report_in_transaction(report_id, scrubbed_by):
+    """See scrub_report for general documentation of what this method does.
+    It's only safe to call this method from within a transaction.
+
+    Args:
+        report_id: str. The id of the model entity to scrub.
+        scrubbed_by: str. The id of the user that is initiating scrubbing of
+            this report, or a const feconf.REPORT_SCRUBBER_BOT_ID if scrubbed
+            by the cron job.
+    """
+    report_entity = app_feedback_report_models.AppFeedbackReportModel.get_by_id(
+        report_id)
+    if not report_entity:
+        raise Exception(
+            'The AppFeedbackReportModel trying to be scrubbed does not '
+            'exist.')
+    if report_entity.platform == (
```

Comment:
Never operate directly on storage models. Construct the domain object immediately and then only operate on it. Only at the end, convert to a storage model and save.

There should be utility functions in each services file for retrieving the relevant domain objects from the model and saving the object. Only those functions should call into the model layer. (See other files in the codebase.)

Reply:
I didn't make this change yet, let me now your thoughts on [this comment](https://github.com/oppia/oppia/pull/12298#discussion_r605053304).

Reply:
Removing the scrubbing functions as per [this thread](https://github.com/oppia/oppia/pull/12298#discussion_r604720590).